### ISSUE TITLE
Assertion failed 'blockNode->OperIsBlk()' in Compiler::impNormStructVal

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/coreclr#15412]
I hit this while trying to implement more efficient statics in CoreRT, but this also has a repro with CoreCLR.

```
.assembly extern mscorlib{}
.assembly cctor{}

// Tests that passing a valuetype load that requires a cctor trigger
// as a method parameter works.

.class MyValue extends [mscorlib]System.ValueType
{
  .field public int32 Value
}

.class Holder
{
  .field public static valuetype MyValue Result at PreInitValue

  .method private hidebysig specialname rtspecialname static 
          void  .cctor() cil managed
  {
    ret
  }
}

.method static int32 Method(valuetype MyValue)
{
  ldarga 0
  ldfld int32 valuetype MyValue::Value
  ret
}

.method static int32 Main(string[] args)
{
  .entrypoint

  ldsfld valuetype MyValue class Holder::Result
  call int32 Method(valuetype MyValue)
  ret
}

.data PreInitValue = int32(100)
```

The problem is that `impNormStructVal` doesn't handle the situation when the child of the `GT_COMMA` is a `GT_FIELD`. We have a `GT_COMMA` because the static field access comes with a `CORINFO_FLG_FIELD_INITCLASS` flag.

**Comment by RussKeldorph** [comment]
@dotnet/jit-contrib

**Comment by briansull** [comment]
Fixed with PR #15475
